src/main/java/org/example/collection/vocabulary/service/impl/CollectionServiceImpl.java

```diff
@@ -0,0 +1,69 @@
+package org.example.collection.vocabulary.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.example.collection.vocabulary.entity.Collection;
+import org.example.collection.vocabulary.exception.InvalidRequestParamException;
+import org.example.collection.vocabulary.model.request.CollectionCreateRequest;
+import org.example.collection.vocabulary.model.request.CollectionUpdateRequest;
+import org.example.collection.vocabulary.model.response.CollectionResponse;
+import org.example.collection.vocabulary.model.response.PageResponse;
+import org.example.collection.vocabulary.repository.CollectionRepository;
+import org.example.collection.vocabulary.service.CollectionService;
+import org.example.collection.vocabulary.utils.mapper.CollectionRequestMapper;
+import org.example.collection.vocabulary.utils.mapper.CollectionResponseMapper;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class CollectionServiceImpl implements CollectionService {
+	private final CollectionRepository collectionRepository;
+
+	@Override
+	public CollectionResponse findById(UUID id) {
+		return CollectionResponseMapper.INSTANCE.map(
+				collectionRepository.findById(id)
+									.orElseThrow(() -> new InvalidRequestParamException("Collection with id " + id + "not found ")));
+	}
+
+	@Override
+	public PageResponse<CollectionResponse> findAll(int pageNumber, int pageSize, UUID userId) {
+		return CollectionResponseMapper.INSTANCE.map(collectionRepository.findAllByUserId(userId.toString(), PageRequest.of(pageNumber,
+																															pageSize)));
+	}
+
+	@Override
+	@Transactional
+	public void save(CollectionCreateRequest request, UUID userId) {
+		Collection collection = CollectionRequestMapper.INSTANCE.map(request);
+		collection.setUserId(userId.toString());
```

Comment:
you dont need to pass user id to the method. use Spring Auditing and the field will be set by SPring